src/views/Events/index.js

```diff
@@ -50,22 +50,108 @@ export default class Events extends Component {
     this.isMobileView = false;
     this.breakpointWidth = 540;
   }
-  componentWillMount() {
+
+  componentDidUpdate() {
+    window.onpopstate = () => {
+      if (!window.location.href.includes('?')) {
+        window.location.reload();
+      } else {
+        this.loadEvents();
+      }
+    };
+  }
+
+  componentDidMount() {
     this.loadEvents();
+    window.addEventListener('resize', this.resize);
   }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resize);
+  }
+
+  async loadPrevious() {
```

Comment:
a comment here explaining whats going on on this particular part would be really helpful.

Reply:
added comment